rfd/0123-tls-routing-behind-layer7-lb.md

```diff
@@ -299,15 +323,56 @@ for different clusters.
 UX is the same as when [TLS Routing is
 enabled](https://github.com/gravitational/teleport/blob/master/rfd/0039-sni-alpn-teleport-proxy-routing.md#local-teleport-proxy).
 
+#### 5 - Client source IPs
+
+A new setting is added to the Proxy service for web APIs to take client IPs from the "X-Forwarded-For" headers:
+```
+proxy_service:
+  # Enables the Proxy service to take client source IPs from the
+  # "X-Forwarded-For" headers for web APIs recevied from layer 7 load balancers
+  # or reverse proxies.
+  # 
+  # IMPORTANT: please make sure the service is behind a layer 7 load balancer or
+  # reverse proxy that sets or appends client IPs in the "X-Forwarded-For"
+  # headers.
+  #
+  # default: false
+  use_xff_header: true
```

Comment:
UX struggle: 
For me this name is quite enigmatic. Firstly a reader need to resolved the xff shortcut and then know a context of `X-Forwarded-For` HTTP header. Wonder if we cannot just name it like `behind_alb: true` 
Also we consider moving this flag to the Teleport Auth settings because we have already `proxy_listener_mode`

Reply:
> Firstly a reader need to resolved the xff shortcut and then know a context of X-Forwarded-For HTTP header. Wonder if we cannot just name it like behind_alb: true
Also we consider moving this flag to the Teleport Auth settings because we have already proxy_listener_mode

I would prefer an individual flag that toggles the behaviour on this header value intake. This flag should not affect proxy listener mode. The flag should be optional as an L7 LB/reverse proxy may not have this header set but Proxy should still function without it (like what is currently released).

What do you think?

As for naming, maybe something like `source_ip_from_lb_headers` is better?

Reply:
can't we enable this option automatically if we detect that we are running behind an L7 lb? There is no other option for them and logs will always be incorrect.

I am wondering if users will enable IP pinning but don't care about the flag option at all and everything will work but we actually don't enforce any IP pinning

Reply:
we had a brief discussion on this [here](https://github.com/gravitational/teleport/pull/26620#discussion_r1202300327). it's not easy to correctly detect if the option needs to be enabled:

1. it's not guaranteed that the LB is setting the IP properly in XFF header, as LB can be in preserve mode for example.
2. we cannot detect this during Proxy start-up since LB may not be healthy yet.

Reply:
I agree that `xff` isn't super clear. Maybe `use_x_forwarded_for` would be better.